contribute.md

```diff
@@ -3,104 +3,345 @@ layout: contribute
-repository.
+In general, the latest stable Node.js should work. However if you are just starting out, you may want to develop against one of the versions that we test against to reduce the chances of you hitting an unexpected issue, although this is not required.
 
-Any version of Node.js that is older than those listed is not supported.
+You can view our [Github Actions configuration](https://github.com/theforeman/foreman/blob/develop/.github/workflows/js_tests.yml) to see supported NodeJS versions. Any version from that list will do. For populating the node_modules folder, npm version 4 or later is required. Alternatively, [yarn](https://yarnpkg.com) can also be used.
 
-For populating the node_modules folder, npm version 4 or later is required. Alternatively,
-[yarn](https://yarnpkg.com) can also be used.
+Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
 
-#### Setup development environment
-* Fork [theforeman/foreman](https://github.com/theforeman/foreman) to your GitHub account. This will create a "foreman" repo under your GitHub username.
-* Clone the fork you just created to your development system:
-{% highlight bash %}
-git clone https://github.com/<username>/foreman.git
-{% endhighlight %}
-* Reference theforeman/foreman as upstream:
-{% highlight bash %}
+```
+dnf module enable nodejs:14
+dnf install nodejs
+```
+
+#### Supported PostgreSQL versions
+
+In general, the latest version of PostgreSQL will work. Minimum version is currently 10, anything newer than that will do the job.
+
+Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+dnf module enable postgresql:10
+dnf install postgresql-server
+```
+
+#### Libraries and database
+
+Some Ruby libraries (rubygems) require development tools (compilers, make), libraries and header files and Chrome driver to be present on the host. Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+dnf groupinstall "Development Tools"
+dnf install libvirt-devel postgresql-devel openssl-devel libxml2-devel sqlite-devel libxslt-devel zlib-devel readline-devel systemd-devel libcurl-devel krb5-devel
+dnf install chromedriver
+```
+
+Foreman requires PostgreSQL, even for development. SQLite is no longer supported and will not work. Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+postgresql-setup --initdb
+```
+
+Configure security, for development purposes you can turn off authentication completely and let all users in on local sockets or loopback network interfaces. If you choose to use passwords, make sure to set them later:
+
+```
+grep trust /var/lib/pgsql/data/pg_hba.conf
+local   all             all                                     trust
+host    all             all             127.0.0.1/32            trust
+host    all             all             ::1/128                 trust
+```
```

Comment:
Or, you use ident auth which is on by default and doesn't require you to change the DB setup. See my suggestions below:

Reply:
Wait, are you saying that it should work without ever touching `pg_hba.conf`?

I am not sure which suggestion below you are referring to.

Reply:
I doublechecked and my clean Fedora installation does not run any ident service on port 113/tcp. I think it is safer to use `trust`.

Reply:
That's a different ident. What I'm saying is that these are the instructions to get it to work:
```bash
$ sudo dnf install postgresql-server -y
$ sudo postgresql-setup --initdb
$ sudo systemctl enable --now postgresql
$ sudo createuser --superuser $USER
$ git clone https://github.com/theforeman/foreman
$ cd foreman
$ bundle install
$ bundle exec rake db:create
$ bundle exec rake db:migrate
$ bundle exec rake db:seed
```

Be sure that https://github.com/theforeman/foreman/commits/2d6f26c2c180e0af9180170f5dcc3ef335c99481 is in what you checkout.